### PR TITLE
Added indexer subsystem

### PR DESCRIPTION
**Pull request by Harry695**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/IndexerSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.RobotConstants.CAN.SparkMax;
+
+@RobotPreferencesLayout(groupName = "Indexer", column = 0, row = 1, width = 2, height = 3)
+public class IndexerSubsystem extends SubsystemBase {
+
+  @RobotPreferencesValue
+  private final RobotPreferences.DoubleValue KS = new RobotPreferences.DoubleValue("Indexer", "KS", 1.0);
```
Comment (edreed) [inline src/main/java/frc/robot/subsystems/IndexerSubsystem.java:27]: The `K` constants shouldn't be `RobotPreferences`. A `KS` of 0.15 is a good start and we can tweak it later. The `KV` and `KA` constants should be calculated from the motor parameters and gear ratio, e.g.

```java
public static double MAX_RPM = MOTOR.getFreeSpeedRPM()/ GEAR_RATIO;
public static double MAX_ACCELERATION = (2 * MOTOR.getStallTorque() * GEAR_RATIO) / MASS;

public static double KV = (RobotConstants.MAX_BATTERY_VOLTAGE - KS) / MAX_RPM;
public static double KA = (RobotConstants.MAX_BATTERY_VOLTAGE - KS) / MAX_ACCELERATION
```

**Comment by Harry695** [comment]
Mr. Reed, the changes you've proposed regarding the feedforward constants have been made.

**Review by edreed — APPROVED** [review]
@Harry695 

LGTM

There was a merge conflict in RobotContainer, so I went ahead and resolved it, You should be good to "Rebase and merge" the PR.

**Comment by edreed** [comment]
@Harry695 Ah, it looks like you merged main at some point. This will prevent the "Rebase and merge". You'll need to "Squash and merge" instead,

**Comment by Space9999** [comment]
Did it for him

**Pull request merged**